pallets/utxo/src/lib.rs

```diff
@@ -230,12 +230,53 @@ pub mod pallet {
         }
     }
 
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    #[derive(Clone, Encode, Decode, Eq, PartialEq, PartialOrd, Ord, RuntimeDebug, Hash)]
+    pub enum TxHeaderAndExtraData {
+        NormalTx {
+            // Normal token ID
+            id: TokenId,
+        },
+        ConfidentialTx, /* not implemented yet */
+        Nft {
+            id: TokenId,
+            data_hash: [u8; 32],
+            data_url: String,
+            creator_pkh: H256,
```

Comment:
Depending on how the implementation will be done, this may also be superfluous. We're gonna keep track of issuance transactions vs token IDs, we can just get this from the issuance transactions. No need to for this to be here.

Reply:
Does it mean that we need only `id`, `data`, and `owner` fields?

Reply:
Why did I add `data_url` here, I'd like to have in the future the possibility like in the ETH to change metadata or link to digital data: https://eips.ethereum.org/EIPS/eip-1155#metadata